stl/inc/xutility

```diff
@@ -3225,25 +3225,33 @@ namespace ranges {
                 return _Last - _First;
             } else {
                 _Adl_verify_range(_First, _Last);
-                auto _UFirst                  = _Get_unwrapped(static_cast<_It&&>(_First));
-                const auto _ULast             = _Get_unwrapped(static_cast<_Se&&>(_Last));
-                iter_difference_t<_It> _Count = 0;
-                for (; _UFirst != _ULast; ++_UFirst) {
-                    ++_Count;
-                }
-
-                return _Count;
+                return static_cast<iter_difference_t<_It>>(

```

Comment:
There are a great many places in the STL that assume that corresponding wrapped-and-unwrapped iterator types have the same associated types. This `static_cast` (ditto 3238) should therefore be unnecessary. If you think we should be paranoid - I'm not bothered - I'd prefer to `static_assert(is_same_v<iter_difference_t<_It>, iter_difference_t<_Unwrapped_t<_It>>>, "YOU NO DO THIS THING!");`.

Reply:
I would not like to see a static_assert because vector<T, small_allocator<T>> fails your test (unwraps to T*), but I also don't mind generating tons of warnings for this case so I would still like to see the static assert be gone.

Reply:
Did I parse that as "No change needed"?

Reply:
My comment: "we shouldn't cast here; you can `static_assert` if you feel it's necessary"
Billy's comment: "we shouldn't cast _or_ `static_assert` here" (The final occurrence of "static assert" is intended to be "static cast".)

TLDR: let's remove the cast.